doc/tutorials/introduction/documenting_opencv/documentation_tutorial.markdown

```diff
@@ -439,6 +439,89 @@ Then include this snippet into documentation:
 compatibility with the old rST documentation. But newly created samples should be included with the
 _snippet_ command, since this method is less affected by the changes in processed file.
 
+### Toggle buttons inclusion commands {#tutorial_documentation_toggle_buttons_commands_include}
+
+Toggle buttons are used to display the selected configuration (e.g. programming language, OS, IDE).
+
+To use the buttons in documentation, _add_toggle_ and _end_toggle_ commands are used.
+
+Example:
+@verbatim
+@add_toggle{Button Name}
+
+  text / code / doxygen commands / ... (Write .markdown like the rest of the file)
+
+@end_toggle
+@endverbatim
+
+For example using toggle buttons with text and [code](@ref tutorial_documentation_commands_include) snippets:
+
+@verbatim
+
+Buttons Example
+---------------
+
+@add_toggle{C++}
+
+   Text for C++ button
+   @snippet samples/cpp/tutorial_code/introduction/documentation/documentation.cpp hello_world
+
+@end_toggle
+
+@add_toggle{Python}
+
+   Text for Python button
+   @snippet samples/python/tutorial_code/introduction/documentation/documentation.py hello_world
+
+@end_toggle
+
+@add_toggle{Java}
+
+   Text for Java button
+   @snippet samples/java/tutorial_code/introduction/documentation/Documentation.java  hello_world
+
+@end_toggle
+@endverbatim
+
+Result looks like this:
+![Buttons with code example](buttons.png)
+<!--
+Due to a doxygen bug:
+https://bugzilla.gnome.org/show_bug.cgi?id=767171
+
+TODO when Doxygen 1.8.12 is released, we can have a dynamic example and not a static image if we:
```

Comment:
Looks like the problem is not in doxygen, but JS code (in footer.html) which incorrectly looks for a header to attach buttons. In this case buttons are being attached to the `h3` header instead of nearest `h2`.